diff --git a/node_modules/cavy-cli/server.js b/node_modules/cavy-cli/server.js
index 64faf22..633013b 100644
--- a/node_modules/cavy-cli/server.js
+++ b/node_modules/cavy-cli/server.js
@@ -2,6 +2,7 @@ const http = require('http');
 const WebSocket = require('ws');
 const chalk = require('chalk');
 const constructXML = require('./src/junitFormatter');
+const { writeFileSync } = require('fs');
 
 // Initialize a server
 const server = http.createServer();
@@ -24,6 +25,9 @@ wss.on('connection', socket => {
     const json = JSON.parse(message);
 
     switch(json.event) {
+      case 'message':
+        logMessage(json.data);
+        break;
       case 'singleResult':
         logTestResult(json.data);
         break;
@@ -62,6 +66,15 @@ function logTestResult(testResultJson) {
   }
 };
 
+function logMessage(json) {
+  const { message, level } = json;
+  switch (level) {
+    case 'log': console.log(chalk.white(message)); break;
+    case 'debug': console.log(chalk.yellow(message)); break;
+    case 'warn': console.log(chalk.bgYellow(message)); break;
+  }
+}
+
 // Internal: Accepts a json report object, console.logs the overall result of
 // the test suite and quits the process with either exit code 1 or 0 depending
 // on whether any tests failed.
@@ -80,6 +93,7 @@ function finishTesting(reportJson) {
   if (server.locals.outputAsXml) {
     constructXML(fullResults);
   }
+  constructMarkdown(fullResults);
 
   // If all tests pass, exit with code 0, else code 42.
   // Code 42 chosen at random so that a test failure can be distinuguished from
@@ -98,4 +112,16 @@ function finishTesting(reportJson) {
   console.log('--------------------');
 };
 
+function constructMarkdown(results) {
+  const filename = 'cavy_results.md';
+  console.log(`Writing results to ${filename}`);
+  const data =
+    `### E2E Test Summary\n` +
+    `|Description 📝|Test results 🧪|Duration ⏰|\n` +
+    `|---|---|---|\n` +
+    results.testCases.map((result) => `|${result.description}|${result.passed ? `✅` : `❌`}|${result.time}s|`).join('\n');
+
+  writeFileSync(filename, data);
+};
+
 module.exports = server;
